src/Components/Toolbar/SaveCharacter/SaveCharacter.tsx

```diff
@@ -1,53 +1,50 @@
-import React from 'react';
-import { setDoc } from '@firebase/firestore';
-import { useContext, useState } from 'react';
+import React, { useContext, useState } from 'react';
+import { DocumentData, DocumentReference } from '@firebase/firestore';
 
 import Button from 'react-bootstrap/Button';
 
-import { UserContext } from '../../context';
-import { addNewCharacterForUser, getUsersSavedCharactersList } from '../Firebase/firebase.utils';
-import { NewCharacterStatsObj } from '../../../types';
+import { UserContext } from '../../../context';
+import {
+  addNewCharacterForUser,
+  createCharacterDocument,
+  getUsersSavedCharacterList
+} from '../../Firebase/firebase.utils';
+import { NewCharacterStatsObj } from '../../../../types';
 import { setId } from 'features/characterSlice';
-import { useAppDispatch, useAppSelector } from 'Components/CustomHooks/reduxHooks';
+import { useAppDispatch, useAppSelector } from 'features/reduxHooks';
 
 interface Props {
-  setShowSaveAlert: React.Dispatch<React.SetStateAction<boolean>>;
+  setShowAlert: React.Dispatch<React.SetStateAction<string | null>>;
 }
 
-const SaveCharacter = ({ setShowSaveAlert }: Props) => {
+const SaveCharacter = ({ setShowAlert }: Props) => {
   const dispatch = useAppDispatch();
   const userContext = useContext(UserContext);
   const user = userContext?.user;
   const userId = user ? user.uid : '';
   const characterName = useAppSelector((state) => state.character.name);
   const selectedAdvantages = useAppSelector((state) => state.character.advantages);
   const selectedDisadvantages = useAppSelector((state) => state.character.disadvantages);
-
   const [isSaving, setIsSaving] = useState(false);
 
   const checkSaveRequirements = async () => {
     const checkIfDuplicate = async () => {
       if (user) {
-        const characterList = await getUsersSavedCharactersList(userId);
+        const characterList = await getUsersSavedCharacterList(userId);
         return characterList.includes(characterName);
       } else {
         return;
       }
     };
 
-    if (selectedAdvantages.length <= 0 && selectedDisadvantages.length <= 0) {
-      console.log('**** Save fail');
-      alert('You must select at least one Advantage or Disadvantage');
-      return;
-    } else if (await checkIfDuplicate()) {
-      console.log('**** Save fail');
-      alert(
-        'You already have a character with this name. Delete original character or change name'
-      );
+    if (await checkIfDuplicate()) {
+      setShowAlert('duplicate');
       return;
     } else if (characterName === '') {
-      console.log('**** Save fail');
-      alert('You must type a name for your character and press enter in order to save');
+      setShowAlert('no-name');
+      return;
+    } else if (selectedAdvantages.length <= 0 && selectedDisadvantages.length <= 0) {
+      setShowAlert('no-attributes');
       return;
```

Comment:
same her, it could just be a fn called saveStatus returning a string of 'no-name' 'no-attibutes' 'success' 

then outside the call could be setShowAlert(saveStatus)